### PR TITLE
Adding delete button to uploads page.

### PR DESCRIPTION
**Pull request by akash1810**
delete-upload

### CODE REVIEW

File: kahuna/public/js/upload/controller.js [hunk 1]
```diff
@@ -1,18 +1,55 @@
...
-}]);
+        // TODO: Show multiple jobs?
+        ctrl.latestJob = uploadManager.getLatestRunningJob();
+
+        // my uploads
+        mediaApi.getSession().then(session => {
+            var uploadedBy = session.user.email;
+            mediaApi.search('', { uploadedBy }).then(resource => {
+
+                resource.data.forEach(image => {
```
Comment (akash1810) [inline kahuna/public/js/upload/controller.js]: :cry:

File: kahuna/public/js/components/gr-delete-image/gr-delete-image.js [hunk 2]
```diff
@@ -10,12 +10,12 @@ deleteImage.controller('grDeleteImageCtrl', ['mediaApi', function (mediaApi) {
         mediaApi.delete(ctrl.image)
             .then((resp) => {
                 if (angular.isDefined(ctrl.onSuccess)) {
-                    ctrl.onSuccess(resp);
+                    ctrl.onSuccess(resp, ctrl.image);
```
Comment (akash1810) [inline kahuna/public/js/components/gr-delete-image/gr-delete-image.js]: need the image so we can do this https://github.com/guardian/grid/pull/1076/files#diff-8357e32a30a84e3c0778b7128dc09891R39

File: kahuna/public/js/upload/controller.js [hunk 3]
```diff
@@ -1,18 +1,55 @@
...
-                  ['$scope', 'uploadManager', 'mediaApi',
-                   function($scope, uploadManager, mediaApi) {
+upload.controller('UploadCtrl', [
+    '$scope', '$state', '$window', 'uploadManager', 'mediaApi',
+    function($scope, $state, $window, uploadManager, mediaApi) {
+        var ctrl = this;
 
-    // TODO: Show multiple jobs?
-    this.latestJob = uploadManager.getLatestRunningJob();
+        ctrl.deletable = new Set();
```
Comment (theefer) [inline kahuna/public/js/upload/controller.js]: Does it need to be exposed on the `ctrl`?

File: kahuna/public/js/upload/controller.js (hunk 1 above)
Comment (theefer) [inline kahuna/public/js/upload/controller.js]: Yeah it's not ideal. Annoying that we can't bind async stuff to templates.

**Comment by theefer** [comment]
:+1:

**Pull request merged**